google/cloud/forseti/scanner/scanners/iam_rules_scanner.py

```diff
@@ -32,6 +35,58 @@
 LOGGER = logger.get_logger(__name__)
 
 
+def _add_bucket_ancestor_bindings(policy_data):
+    """Add bucket relevant IAM policy bindings from ancestors.
+
+    Resources can inherit policy bindings from ancestors in the resource
+    manager tree. For example: a GCS bucket inherits a 'objectViewer' role
+    from a project or folder (up in the tree).
+
+    So far the IAM rules engine only checks the set of bindings directly
+    attached to a resource (direct bindings set (DBS)). We need to add
+    relevant bindings inherited from ancestors to DBS so that these are
+    also checked for violations.
+
+    If we find one more than one binding with the same role name, we need to
+    merge the members.
+
+    NOTA BENE: this function only handles buckets and bindings relevant to
+    these at present (but can and should be expanded to handle projects and
+    folders going forward).
+
+    Args:
+        policy_data (list): list of (parent resource, iam_policy resource,
```

Comment:
is `parent resource` really the first element of the tuple?

Reply:
Yes, indeed.

Reply:
I am not seeing this is really a `parent resource`.  This ```policy_data``` comes directly from `_retrieve()` and it looks like _a `resource` with parent information inside.  But the ```data```attribute is that of the current resource, not parent.